src/foam/core/Validation.js

```diff
@@ -354,30 +354,19 @@ foam.CLASS({
       name: 'validationPredicates',
       factory: function() {
         var self = this;
-        var ret = [
+        if ( ! this.required ) return;
```

Comment:
```suggestion
        if ( ! this.required ) return [];
```

I think we need to return an empty array instead of undefined in this case. The code that uses the validationPredicates expects it to be an array. That's what the other validationPredicates definitions in this file do as well. Sorry I didn't catch this the first time.